### ISSUE TITLE
Inconsistency `CrossEntropyLoss` vs `BCELoss` regarding logits/probability space

### ISSUE THREAD

**Issue by ego-thales** [pytorch/pytorch#128493]
### 🚀 The feature, motivation and pitch

The loss [`CrossEntropyLoss`](https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html#torch.nn.CrossEntropyLoss) expects logits inputs. On the other hand, there exist [`BCELoss`](https://pytorch.org/docs/stable/generated/torch.nn.BCELoss.html#torch.nn.BCELoss) and [`BCEWithLogitsLoss`](https://pytorch.org/docs/stable/generated/torch.nn.BCEWithLogitsLoss.html#torch.nn.BCEWithLogitsLoss). But there is no counterpart to `CrossEntropyLoss` that expects inputs in the probability space.

### Alternatives

I think it would make more sense if `CrossEntropyLoss` expected probability inputs and if a new `CrossEntropyWithLogitsLoss` was introduced. But this would be a dramatic change in the API.

As such, I would propose to introduce an option `with_logits` (that defaults to `True` to keep compatibility) to `CrossEntropyLoss`.

What do you think?

### Additional context

Finally, I ask the following: is it possible to easily train a neural net that outputs softmax probabilities and not logits? It seems like since there is no `CrossEntropyLoss` counterpart for probability space, the user is forced to add a extra step  and use `NLLLoss` or write a custom loss.

Thanks!

**Comment by rajveer43** [comment]
Can I Work on it?

**Comment by jbschlosser** [comment]
Note that [`nn.NLLLoss`](https://pytorch.org/docs/stable/generated/torch.nn.NLLLoss.html) takes in inputs as log-probabilities, and operating on these is generally more numerically stable.

> Is the fact that CrossEntropyLoss doesn't work with probability space due to under/overflow handling in general?

So I'd say yes to this :)
